buffer/src/test/java/io/netty/buffer/ReadOnlyDirectByteBufferBufTest.java

```diff
@@ -423,4 +425,55 @@ public void testMemoryAddress() {
             buf.release();
         }
     }
+
+    @Test
+    public void testDuplicate() {
+        ByteBuf buf = buffer(allocate(8).asReadOnlyBuffer());
+        try {
+            assertEquals(buf.getClass(), buf.duplicate().getClass());
```

Comment:
I don't think we should assert on the concrete classes, but rather that the resulting buffers are read-only and contain the expected contents.

Reply:
Fixed PTAL again